samples/openthread/coap_client/src/coap_client_utils.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <coap_server_client_interface.h>
+#include <coap_utils.h>
+#include <logging/log.h>
+#include <net/openthread.h>
+#include <net/socket.h>
+#include <openthread/thread.h>
+
+#include "coap_client_utils.h"
+
+LOG_MODULE_REGISTER(coap_client_utils, CONFIG_COAP_CLIENT_UTILS_LOG_LEVEL);
+
+#define RESPONSE_POLL_PERIOD 100
+
+static u32_t poll_period;
+
+static struct k_work unicast_light_work;
+static struct k_work multicast_light_work;
+static struct k_work toggle_MTD_SED_work;
+static struct k_work provisioning_work;
+static struct k_work on_connect_work;
+static struct k_work on_disconnect_work;
+
+mtd_mode_toggle_cb_t on_mtd_mode_toggle;
+
+/* Options supported by the server */
+static const char *const light_option[] = { LIGHT_URI_PATH, NULL };
+static const char *const provisioning_option[] = { PROVISIONING_URI_PATH,
+						   NULL };
+
+/* Thread multicast mesh local address */
+static struct sockaddr_in6 multicast_local_addr = {
+	.sin6_family = AF_INET6,
+	.sin6_port = htons(COAP_PORT),
+	.sin6_addr.s6_addr = { 0xff, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			       0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01 },
+	.sin6_scope_id = 0U
+};
+
+/* Variable for storing server address acquiring in provisioning handshake */
+static char unique_local_addr_str[INET6_ADDRSTRLEN];
+static struct sockaddr_in6 unique_local_addr = {
+	.sin6_family = AF_INET6,
+	.sin6_port = htons(COAP_PORT),
+	.sin6_addr.s6_addr = {0, },
+	.sin6_scope_id = 0U
+};
+
+static struct otInstance *openthread_get_default_instance(void)
```

Comment:
Is this still needed? Isn't that already implemented in the ongoing zephyr upmerge?

Reply:
After upmerge finish, we should it remove.

Reply:
Should be removed for a final PR - I've just removed it from existing samples on an upmerge branch.

Reply:
Thank you, I know that.